pkg/machine/fedorawsl.go

```diff
@@ -0,0 +1,65 @@
+//go:build amd64 || arm64
+// +build amd64 arm64
+
+package machine
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	githubLatestReleaseURL = "https://github.com/gbraad/podman-wsl-fedora/releases/latest/download/rootfs.tar.xz"
+)
+
+func NewFedoraWSLDownloader(vmType, vmName, releaseStream string) (DistributionDownload, error) {
+	// TODO:
+	imageName, downloadURL, size, err := getFedoraWSLDownload(githubLatestReleaseURL)
+	if err != nil {
+		return nil, err
+	}
+
+	dataDir, err := GetDataDir(vmType)
+	if err != nil {
+		return nil, err
+	}
+
+	f := FedoraDownload{
+		Download: Download{
+			Arch:      getFcosArch(),
+			Artifact:  artifact,
+			Format:    Format,
+			ImageName: imageName,
+			LocalPath: filepath.Join(dataDir, imageName),
+			URL:       downloadURL,
+			VMName:    vmName,
+			Size:      size,
+		},
+	}
+	f.Download.LocalUncompressedFile = f.getLocalUncompressedName()
+	return f, nil
+}
+
+func getFedoraWSLDownload(releaseURL string) (string, *url.URL, int64, error) {
+	resp, err := http.Head(releaseURL)
+
+	downloadURL, err := url.Parse(releaseURL)
+	if err != nil {
+		return "", nil, -1, errors.Wrapf(err, "invalid URL generated from discovered Fedora file: %s", releaseURL)
+	}
+
+	if err != nil {
+		return "", nil, -1, errors.Wrapf(err, "head request failed: %s", releaseURL)
+	}
+	_ = resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", nil, -1, fmt.Errorf("head request failed [%d] on download: %s", resp.StatusCode, releaseURL)
+	}
+
+	return "podman-wsl-f35.tar.gz", downloadURL, resp.ContentLength, nil
```

Comment:
I'd drop the release version since we can technically update the same location with a new base. (Also currently it's based on f36 now)

Reply:
`imageName` is static in this case: `rootfs.tar.xz`